Foxtrot/Driver/Rewriting/Utils/TypeNodeExtensions.cs

```diff
@@ -0,0 +1,20 @@
+﻿using System.Compiler;
+using System.Diagnostics.Contracts;
+
+namespace Microsoft.Contracts.Foxtrot.Utils
+{
+    internal static class TypeNodeExtensions
+    {
+        public static int TemplateArgumentsCount(this TypeNode typeNode)
```

Comment:
The same as before. But this time I prefer this extension method over your suggested solution. 

Extension method is much more descriptive and high level. Proposed solution required "slow memory" to analyze, IMO.